From e20ac714a949820660e9aa10aad03c791df7caa9 Mon Sep 17 00:00:00 2001
From: effects3d <effects3dee@gmail.com>
Date: Thu, 27 Aug 2026 00:29:12 +0200
Subject: [PATCH 3/9] =?UTF-8?q?Partially=20revert=20=E2=80=9Cwinex11:=20Li?=
 =?UTF-8?q?sten=20for=20ConfigureNotify=20events=20on=20the=20virtual=20de?=
 =?UTF-8?q?sktop.=E2=80=9D?=
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Reverts a regression with Qt menus turning invisible after interacting
with a child window in virtual desktop mode. Annoyingly enough, there’s
still a chance of that occuring, but it should be a lot less frequent
now.

Source: https://gitlab.winehq.org/wine/wine/-/commit/06b1a49f
Signed-off-by: effects3d <effects3dee@gmail.com>
---
 dlls/winex11.drv/desktop.c | 2 +-
 dlls/winex11.drv/window.c  | 4 ++++
 2 files changed, 5 insertions(+), 1 deletion(-)

diff --git a/dlls/winex11.drv/desktop.c b/dlls/winex11.drv/desktop.c
index 3c223d66b9b..406f5194fba 100644
--- a/dlls/winex11.drv/desktop.c
+++ b/dlls/winex11.drv/desktop.c
@@ -77,7 +77,7 @@ BOOL X11DRV_CreateDesktop( const WCHAR *name, UINT width, UINT height )
     /* Create window */
     win_attr.event_mask = ExposureMask | KeyPressMask | KeyReleaseMask | EnterWindowMask |
                           PointerMotionMask | ButtonPressMask | ButtonReleaseMask | FocusChangeMask |
-                          StructureNotifyMask | PropertyChangeMask;
+                          PropertyChangeMask;
     win_attr.cursor = XCreateFontCursor( display, XC_top_left_arrow );
 
     if (default_visual.visual != DefaultVisual( display, DefaultScreen(display) ))
diff --git a/dlls/winex11.drv/window.c b/dlls/winex11.drv/window.c
index ce37a4c4d5e..3fdfe3b15d5 100644
--- a/dlls/winex11.drv/window.c
+++ b/dlls/winex11.drv/window.c
@@ -1633,6 +1633,10 @@ static void window_set_config( struct x11drv_win_data *data, RECT rect, BOOL abo
     TRACE( "window %p/%lx, requesting config %s mask %#x above %u, serial %lu\n", data->hwnd, data->whole_window,
            wine_dbgstr_rect(new_rect), mask, above, data->configure_serial );
     XReconfigureWMWindow( data->display, data->whole_window, data->vis.screen, mask, &changes );
+
+    /* don't expect a ConfigureNotify while window is unmapped */
+    if (data->pending_state.wm_state == WithdrawnState) data->configure_serial = 0;
+
     if (mask & (CWWidth | CWHeight)) clear_emulated_fullscreen_padding( data );
 
     if (net_wm_state != -1) window_set_net_wm_state( data, net_wm_state );
-- 
2.55.0

